projects/client/RabbitMQ.Client/src/client/impl/ConsumerWorkService.cs

```diff
@@ -98,14 +77,13 @@ void Loop()
                         {
                         }
                     }
-
-                    messageArrived.WaitOne(waitTime);
```

Comment:
No need to poll anymore, since the Semaphore + WaitAsync takes care of suspending the task until it has something to do. Better core utilization.